src/metabase/query_processor/streaming/xlsx.clj

```diff
@@ -428,73 +429,76 @@
   "Adds a row of values to the spreadsheet. Values with the `scaled` viz setting are scaled prior to being added.
 
   This is based on the equivalent function in Docjure, but adapted to support Metabase viz settings."
-  {:arglists '([sheet values cols col-settings cell-styles typed-cell-styles])}
-  (fn [sheet _values _cols _col-settings _cell-styles _typed-cell-styles]
+  {:arglists '([sheet values cols col-settings cell-styles typed-cell-styles]
+               [sheet row-num values cols col-settings cell-styles typed-cell-styles])}
```

Comment:
add an arity here that adds `row-num` as an arg. 

This is used in the pivot table case because the pivot initialization function needs to create an empty row to attach the pivot header sources. If we don't explicitly target that row number, all rows will be added in an append style, and be off by 1, and the pivot table will work but not have the heading labels.